spec/jobs/order_cycle_webhook_job_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe OrderCycleWebhookJob do
+  let(:endpoint) { WebhookEndpoint.create url: "http://url", subscriptions: "order_cycle.event" }
+  let(:order_cycle) {
+    create(
+      :order_cycle,
+      name: "Order cycle 1",
+      orders_open_at: "2022-09-19 09:00:00 +1000".to_time,
+      orders_close_at: "2022-09-19 17:00:00 +1000".to_time,
+    ).tap { |oc|
+      oc.coordinator.update! name: "Starship Enterprise", webhook_endpoint: endpoint
+    }
+  }
+
+  it "creates webhook payload for specified order cycle only" do
+    order_cycle.dup.save # Ensure another OC is not called
+
+    data = {
+      id: order_cycle.id,
+      name: "Order cycle 1",
+      orders_open_at: "2022-09-19 09:00:00 +1000",
+      orders_close_at: "2022-09-19 17:00:00 +1000",
+      enterprise_id: order_cycle.coordinator_id,
+      enterprise_name: "Starship Enterprise",
+    }
+    expect(WebhookDeliveryJob).to receive(:perform_later)
+      .with("http://url", "order_cycle.event", hash_including(data)).once
+
+    OrderCycleWebhookJob.perform_now(order_cycle.id, 'order_cycle.event')
+  end
+
+  it "creates webhook payload when no open or close times specified" do
+    order_cycle.update!(orders_open_at: nil, orders_close_at: nil)
+    data = {
+      orders_open_at: nil,
+      orders_close_at: nil,
+    }
+    expect(WebhookDeliveryJob).to receive(:perform_later)
+      .with("http://url", "order_cycle.event", hash_including(data))
+
+    OrderCycleWebhookJob.perform_now(order_cycle.id, 'order_cycle.event')
+  end
+
+  context "not subscribed to an endpoint" do
+    let(:order_cycle) { create(:order_cycle) }
+
+    it "doesn't create webhook payload" do
+      expect(WebhookDeliveryJob).to_not receive(:perform_later)
+
+      OrderCycleWebhookJob.perform_now(order_cycle.id, 'order_cycle.event')
+    end
+  end
+
+  context "subscribed to different event" do
+    it "doesn't create webhook payload" do
+      expect(WebhookDeliveryJob).not_to receive(:perform_later)
+        .with("http://url", any_args)
```

Comment:
I guess that this would pass if we accidentally enqueued a job with a different URL? Shall we test for enqueued jobs instead?

Reply:
Mocking or testing for method calls is always a bit brittle. Better to test for outcomes (job in the database).

Reply:
> I guess that this would pass if we accidentally enqueued a job with a different URL? 

Yeah, I think we should test that no jobs were created, regardless of url. Updated.

> Better to test for outcomes (job in the database).

Totally. I was having trouble and gave up before. Don't know why though, it seems obvious now. Done.